### PR TITLE
[393] Implement Exhibit States

### PR DESCRIPTION
**Pull request by arnauddieumegard**
- Implement M2 methods
- Add "exhibit states" compartment on STV + creation tools
- Add "exhibit states" related tools on definition, usage nodes and the "state transition" compartment
- Provide "exhibit states" multiline label on ExhibitState
- Add abstract getSemanticCandidatesExpression on common Abstract[Definition|Usage]NodeDescriptionProvider
- Add common ViewToolDescriptionService class to hold descriptions tool related services. Used in DiagramDescriptionProviders.
- Minor code refactorings (AQLUtils)

Bug: https://github.com/eclipse-syson/syson/issues/393

### CODE REVIEW

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ViewToolDescriptionService.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import org.eclipse.syson.util.AQLUtils;
+import org.eclipse.syson.util.IDescriptionNameGenerator;
+import org.eclipse.syson.util.SysMLMetamodelHelper;
+
+/**
+ * Services class for building View Description.
+ *
+ * @author adieumegard
+ */
+public class ViewToolDescriptionService {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ViewToolDescriptionService.java:33]: May be `ToolDescriptionService` would be enough? What do you think about?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ViewToolDescriptionService.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    protected final DiagramBuilders diagramBuilderHelper = new DiagramBuilders();
+
+    /**
+     * Create a {@link NodeTool} adding/displaying existing elements from {@code self} on the context diagram or diagram
+     * element.
+     *
+     * @param recursive
+     *            Whether or not the created tool is recursive
+     * @return
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ViewToolDescriptionService.java:45]: Please add javadoc for `@return`

File: backend/services/syson-services/src/main/java/org/eclipse/syson/services/UtilService.java [hunk 3]
```diff
@@ -112,6 +115,29 @@ public Type getReceiverContainerDefinitionOrUsage(AcceptActionUsage aau) {
         return null;
     }
 
+    /**
+     * Get the AQL service expression getting all reachable {@link StateUsage} elements which are not referential
+     * ExhibitStates.
+     *
+     * @param domainType
+     *            A type to be converted as an EClass name
+     * @return
```
Comment (AxelRICHARD) [inline backend/services/syson-services/src/main/java/org/eclipse/syson/services/UtilService.java:124]: Please add javadoc for `@return` here and everywhere else in this PR

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/diagram/AbstractDiagramDescriptionProvider.java [hunk 4]
```diff
@@ -12,32 +12,34 @@
...
  * @author arichard
  */
-public abstract class AbstractDiagramDescriptionProvider extends AbstractViewElementDescriptionProvider implements IRepresentationDescriptionProvider {
+public abstract class AbstractDiagramDescriptionProvider implements IRepresentationDescriptionProvider {
+
+    protected final ViewBuilders viewBuilderHelper = new ViewBuilders();
+
+    protected final DiagramBuilders diagramBuilderHelper = new DiagramBuilders();
+
+    protected final ViewToolDescriptionService viewDescriptionToolService = new ViewToolDescriptionService();
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/diagram/AbstractDiagramDescriptionProvider.java:35]: viewDescriptionToolService => toolDescriptionService

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/nodes/ExhibitStatesCompartmentItemNodeDescriptionProvider.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package org.eclipse.syson.diagram.common.view.nodes;
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/nodes/ExhibitStatesCompartmentItemNodeDescriptionProvider.java:1]: Copyright is missing

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ViewToolDescriptionService.java (hunk 1 above)
Comment (arnauddieumegard) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ViewToolDescriptionService.java:33]: Yes. "View" comes from the fact that we use the ViewBuilders in this class. But I can definitely remove the View if you want.

Comment (AxelRICHARD) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ViewToolDescriptionService.java:33]: ok, let's make it short and simple

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**